src/birdtrayapp.cpp

```diff
@@ -0,0 +1,148 @@
+#include <QtCore/QLibraryInfo>
+#include "birdtrayapp.h"
+#ifdef Q_OS_WIN
+#  include "birdtrayeventfilter.h"
+#endif /* Q_OS_WIN */
+#include "utils.h"
+#include "morkparser.h"
+
+
+BirdtrayApp::BirdtrayApp(int &argc, char** argv) : QApplication(argc, argv) {
+    QApplication::setWindowIcon(QIcon(QString::fromUtf8(":/res/birdtray.ico")));
+    QCoreApplication::setOrganizationName("ulduzsoft");
+    QCoreApplication::setOrganizationDomain("ulduzsoft.com");
+    QCoreApplication::setApplicationName("birdtray");
+    QCoreApplication::setApplicationVersion(Utils::getBirdtrayVersion());
+    // This prevents exiting the application when the dialogs are closed on Gnome/XFCE
+    QApplication::setQuitOnLastWindowClosed(false);
+    
+    bool translationLoadedSuccessfully = loadTranslations();
+    QCoreApplication::installTranslator(&qtTranslator);
+    QCoreApplication::installTranslator(&dynamicTranslator);
+    QCoreApplication::installTranslator(&mainTranslator);
+
+#ifdef Q_OS_WIN
+    BirdtrayEventFilter filter;
+    installNativeEventFilter(&filter);
+#endif /* Q_OS_WIN */
+    
+    QCommandLineParser parser;
+    parseCmdArguments(parser);
+    
+    QString morkPath = parser.value("dump-mork");
+    if (!morkPath.isEmpty()) {
+        MorkParser::dumpMorkFile(morkPath);
+        exit(1); // TODO: Why 1? Replace with return code of dumpMorkFile
```

Comment:
This was copied from `main.cpp`. Is there a reason why we return 1 (`EXIT_FAILURE`) and not the 0 returned by `MorkParser::dumpMorkFile`?

Reply:
Fine with me.

Reply:
Sorry, I'm not sure if you meant that you are ok with the current way, or if you are ok with changing these. If it is the first, we should remove the TODOs, if it is the second, do you want me to create the changes or are you on it?